### DISCUSSION TITLE
Link Component behaves weird

### DISCUSSION THREAD

**Discussion by hannarosenfeld** [gatsbyjs/gatsby#35613] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Hi, I am having issue understanding this.
I am mapping over a graphql query for an image slider.
Each image contains next to the image, a text, and also a link.
The link leads to a specific, individual page of the website. Or at least, it should.

### Reproduction Link

https://github.com/hannarosenfeld/leo-leonhard-contentful

### Steps to Reproduce

```
export default class Slidy extends Component {
    render() {
        return (
            <StaticQuery
              query={graphql`
                  query SlickQuery {
                    allContentfulSliderImage {
                      nodes {
                        page
                        imageTitle
                        id
                        sliderImage {
                          gatsbyImageData(
                          width: 1200,
                          )
                        }
                      }
                    }
                  }
               `}
              render={data => (
                  <div>
                    <Slider {...settings}>
         {data.allContentfulSliderImage.nodes.map((image) => {
        const slickImage = getImage(image.sliderImage)
                          return(
                              <div key={image.id}>
                                <BgImage
                                  alt={image.imageTitle}
                                  image={slickImage}
                                >
                                  <Link to={image.page}>
                           <div>{image.imageTitle</div>
                                  </Link>
                                </BgImage>
                              </div>
                          )
                      })}
                    </Slider>
                  </div>
              )}
            />

        );
    }
}
```

### Expected Result

Clicking on an image should lead me to the corresponding page based on the  graphql query. Checking Graphiql, this should work!!
![Screen Shot 2022-05-09 at 8 12 31 AM](https://user-images.githubusercontent.com/22946092/167412723-b4c7310c-410a-44a3-90ee-cc9fcce29256.png)

### Actual Result

The problem is, it will always link to the same page. 

`<Link to={image.page} />`

It will always link to the last value for `image.page` ("publikationen", see query) Why?

ps, doing something like `<a href={image.page}>{image.page}</a>` will show me the right value, like let's say "grafik" for the image that should link to the "grafik" page - but then link to "publikationen" anyway.
This seems a specific issue to the `<a/>` and `<Link/>` tags.

### Environment

```shell
System:
    OS: macOS 10.15.7
```

**Comment by LekoArts** [comment]
Hi!

I haven't checked out your code or reproduction in detail yet but from what you provide so far it seems like your links resolve to something like

```jsx
<Link to="publikationen" />
```

`gatsby-link` uses Reach Router under the hood and this syntax resolves to a "relative link": https://reach.tech/router/nesting

So if e.g. you're on `/foo` and use `to="bar"` you'll be routed to `/foo/bar`.

What happens when you use?

```jsx
<Link to={`/${image.page}`} />
```

To have a leading slash in front of the names so that it becomes `/publikationen`

**Comment by hannarosenfeld** [comment]
Hi, thank you. When I use ```<Link to={`/${image.page}`} />``` the exact same thing happens. Also when I use it within an `<a/>` tag.

**Comment by LekoArts** [comment, marked as answer]
It's this issue: https://github.com/akiran/react-slick/issues/2076

I've added

```css
.slick-slide:not(.slick-active) {
    pointer-events: none;
}
```

to your `src/styles/styles.css` and it works now. I found it through searching for `anchor`, `link` in the repository. Always good to check out the GitHub of the project you use :)

**Reply by hannarosenfeld** [reply]
Wow, thank you so so much!!!! I would have never figured this out.
